File: src/019. Counting Sundays/019.rb
require 'date'

first_of_month = Date.new(1901, 1, 1)
count = 0
while first_of_month < Date.new(2000, 12, 31)
  count += 1 if first_of_month.sunday?
  first_of_month = first_of_month.next_month
end

puts count

File: src/019. Counting Sundays/019.go
package main

import (
	"fmt"
	"time"
)

func main() {
	firstOfMonth := time.Date(1901, 1, 1, 0, 0, 0, 0, time.UTC)
	count := 0
	for firstOfMonth.Before(time.Date(2000, 12, 31, 0, 0, 0, 0, time.UTC)) {
		if firstOfMonth.Weekday() == time.Sunday {
			count += 1
		}
		firstOfMonth = firstOfMonth.AddDate(0, 1, 0)
	}
	fmt.Println(count)
}
